include/deal.II/lac/linear_operator.h

```diff
@@ -169,11 +170,12 @@ template <typename Range, typename Domain> class LinearOperator
   LinearOperator (const LinearOperator<Range, Domain> &) = default;
 
   /**
-   * Templated copy constructor that creates a LinearOperator object from an
-   * object @p op for which the conversion function
+   * Templated copy constructor that creates a LinearOperator object from
+   * an object @p op for which the conversion function
    * <code>linear_operator</code> is defined.
    */
-  template<typename Op>
+  template<typename Op,
+           typename = typename std::enable_if<!std::is_base_of<LinearOperator<Range, Domain>, Op>::value>::type>
   LinearOperator (const Op &op)
   {
     *this = linear_operator<Range, Domain, Op>(op);
```

Comment:
Or, to avoid recursion with the assignment operator below, something like
```
  *this = static_cast<const LinearOperator<Range,Domain>&>(op);
```
(At least one of the two needs to have the static_cast.)